README.md
# Short Term Network Web (STNWeb) v2

[![build status](https://github.com/USGS-WiM/stnweb2/workflows/Build/badge.svg)](https://github.com/coryrylan/angular-github-actions/actions) [![GitHub Super-Linter](https://github.com/USGS-WiM/stnweb2/workflows/Lint%20Code%20Base/badge.svg)](https://github.com/marketplace/actions/super-linter)

![WiM](wimlogo.png)

2nd generation of STNWeb, the data entry and data management web application for the USGS Short Term Network high-water mark and storm tide sensor database.

## Prerequisites

[Node](https://nodejs.org/en/),[NPM](https://www.npmjs.com/), [Angular CLI](https://cli.angular.io/)

```bash
npm install -g @angular/cli
```

Recommended node version: v12.14 or greater.

## Getting Started

These instructions will get you a copy of the project up and running on your local machine for development and testing purposes. See deployment for notes on how to deploy the project on a live system.

### Installing

Clone the repo and `cd` to that directory

install dependencies:

```bash
npm install
```

### Dev server

Serve application with live reload change detection:

```bash
ng serve
```

### Building and testing

To build a distribution folder for production:

```bash
ng build --prod --base-href --aot=false --build-optimizer=false
```

To run tests:
```bash
ng test --watch=false --browsers=ChromeHeadlessCustom
```

### Development Workflow

**Instructions below are for developers actively working on the repo**

An issue will be assigned to you via Github. Your workflow begins after assignment:

1. Create a branch with the issue number as the branch name (e.g. `134`)
2. Move issue into 'In Progress' column of project board
3. Do the work
    - While you work, you may wish to have the app running with live reload. Run `ng serve` to do so.
4. Write tests for the work
    - This is a critical step. STNWeb2 uses the [Jasmine library](https://jasmine.github.io/) for unit testing. Please refer there for docs and examples. Also refer to the [Angular testing guide](https://angular.io/guide/testing).
5. Run the tests to be sure they all pass and that the overall thresholds are met: `ng test`
6. Be sure the app runs in the browser without errors: `ng serve`
7. Ensure the app builds without error: `ng build`
    - If any of the checks above fail, please fix the issue. Ask for help if needed.
8. Once all checks have passed and you are ready to submit a Pull Request, update the changelog with a brief description of what your work added, changed, or fixed. There is an Unreleased section at top with subheadings for each category. Edit the file CHANGELOG.md found at project root.
9. Add the changed files `git add .` and commit `git commit -m '[your commit message here]'` you commit message should reference the issue number and include a very brief description of the work.
10. **Pull from `dev`**
    - Run `git pull origin dev`. This is a critical step. It ensures your Pull Request is synced with the latest work in the main `dev` branch. If you are lucky, it will auto-merge. Otherwise, you may have to resolve conflicts between your commit and what currently exists in dev. Please be careful with this step so no code is lost - ask for help if you are unsure what to do.
    - If manually merging, you will have changed files so you will need to add and commit once more (see step 9).
11. Push your committed and synced branch to the remote repo (Github): `git push origin [your branch name]`
12. Submit Pull Request (PR) to merge your issue branch into `dev`
    - Automated checks will be run against your PR. Failure of any of these will mean you need to re-commit to your branch to update your PR with the fix. A colleague will review your work and either approve it or request changes. Upon approval of the PR your issue will be automatically moved to the 'In Dev Branch' column of the project board.

Move onto next assigned issue and start back at step 1.

**Work will be merged with Master when sprint is complete for a release.**

### Deployment

TBD

## Built With

-   [Angular](https://angular.io/) - The main web framework used
-   [NPM](https://www.npmjs.com/) - Dependency Management

## Contributing

Please read [CONTRIBUTING.md](CONTRIBUTING.md) for details on the process for submitting pull requests to us. Please read [CODE_OF_CONDUCT.md](CODE_OF_CONDUCT.md) for details on adhering by the [USGS Code of Scientific Conduct](https://www2.usgs.gov/fsp/fsp_code_of_scientific_conduct.asp).

## Versioning

We use [SemVer](http://semver.org/) for versioning. For the versions available, see the [tags on this repository](../../tags).

## Changelog

All developers must maintain the Changelog located at root as CHANGELOG.md. This document serves as a portable history of changes to the codebase and is meant to complement Github release tags. 'Added', 'Changed', and 'Fixed' sections should be updated within the Unreleased header when creating a Pull Request. As releases are created, those sections will move down into respective release headers.

## Authors

Lauren Privette - _Lead Developer_ - [USGS Web Informatics & Mapping](https://wim.usgs.gov/)

Blake Draper - _Developer_ - [USGS Web Informatics & Mapping](https://wim.usgs.gov/)

Abby Gleason - _Developer_ - [USGS Web Informatics & Mapping](https://wim.usgs.gov/)

See also the list of [contributors](../../graphs/contributors) who participated in this project.

## License

This project is licensed under the Creative Commons CC0 1.0 Universal License - see the [LICENSE.md](LICENSE.md) file for details

## Suggested Citation

In the spirit of open source, please cite any re-use of the source code stored in this repository. Below is the suggested citation:

`This project contains code produced by the Web Informatics and Mapping (WIM) team at the United States Geological Survey (USGS). As a work of the United States Government, this project is in the public domain within the United States. https://wim.usgs.gov`

## Acknowledgments

## About WIM

-   This project authored by the [USGS WIM team](https://wim.usgs.gov)
-   WIM is a team of developers and technologists who build and manage tools, software, web services, and databases to support USGS science and other federal government cooperators.
-   WIM is a part of the [Upper Midwest Water Science Center](https://www.usgs.gov/centers/wisconsin-water-science-center).


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
## [Unreleased](https://github.com/USGS-WiM/stnweb2/tree/dev)

### Added

 -  Add ability to add, edit, delete, reupload and view file details in HWM Edit modal
 -  Loading spinner in HWM edit modal when files are submitted
 -  Add ability to add, edit, delete, reupload and view file details in Reference Datum Edit modal
 -  HWM Label column in HWM Files table in files tab
 -  Ability to change file type when editing a file in files tab
 -  Ability to view file details in Site Edit modal
 -  Dialog warning users that changes will not be saved when a file is being edited and cancel is clicked in all entity edit modals

### Changed

 -  Removed Link filetype options from all file edit/add dialogs
 -  File edit, add and delete styling in Site Edit modal
 
 ### Fixed
 
 -  Unable to upload file after clicking one file type then switching to another type in HWM Edit modal
 -  Site page changing to All Events on page refresh
 -  Unable to change file type to photo when editing a file in sensor edit modal

## [v0.8.1](https://github.com/USGS-WiM/stnweb2/releases/tag/v0.8.1) - 2022-06-24

### Added

 -  Clear 'X' button and progress spinner to event state and event type filters
 -  Info tooltips to fields in edit/add modals where units are converted on submit
 -  Active highlighting on hover, button click and row add to site page tables
 -  When new row item is added and table page is full, jump to next page
 -  Active highlighting on hover and details button click to map filter results table
 -  Ability to delete landowner when editing a site
 -  Ability to add sensor file, datafile and NWIS file in files tab
 -  Ability to delete sensor, data and NWIS files in files tab
 -  Ability to edit sensor, data and nwis files in file tab

### Changed

 -  Disabled sensor retrieve button when no event is selected
 -  Active highlighting style
 -  Fade out active highlight when row is added
 -  Site edit modal arrangement and styling
 
 ### Fixed
 
 -  Fields not populating in details modals
 -  Unedited rows removed from reference datum table after edit
 -  Update counts in site page tabs on edit/add/delete
 -  Validation error incorrectly shown when date changed in sensor retrieve modal
 -  UTC Preview not updating when minutes changed in Sensor add/edit
 -  Update counts in file tab and site file expansion panel when files added/deleted
 -  Date/time formatting and time zone conversion for all entities
 -  Broken USWDS banner images

## [v0.8.0](https://github.com/USGS-WiM/stnweb2/releases/tag/v0.8.0) - 2022-06-05

### Added

 -  Ability to create HWMs
 -  Dropdown functionality to state and event filters on map page
 -  Added ability to delete HWMs
 -  Ability to create Reference Datums
 -  Added ability to delete reference datums
 -  Added ability to deploy a sensor
 -  Added ability to delete sensors
 -  Alert for deploying and retrieving sensors when no event selected
 -  Added ability to create a peak
 -  Added tooltips to all add button icons
 -  Added ability to delete a peak
 -  Added Description and Is Destroyed columns in reference datum table
 -  Added ability to retrieve a sensor
 
### Changed

 -  Clear filters button styling to link
 -  Increased details modal widths
 -  Improved responsiveness of tables in details modals
 -  Autopopulate source and agency when adding file in site edit form
 -  Disabled add button when no event selected for Peaks, Sensors, HWMs and files (except site files)
 -  Disabled peak edit when no event selected
 -  Date Recovered to Date Last Checked in reference datum detail and edit modals
 
 ### Fixed
 
 -  Time reset in sensor edit modal when Cancel Edits is clicked
 -  Time validation in sensor edit
 -  Inability to login after logging out
 -  Duplicated disabled buttons when incorrect role and no event selected

## [v0.7.0](https://github.com/USGS-WiM/stnweb2/releases/tag/v0.7.0) - 2022-01-17

### Added

-   Edit ability for Peaks
-   View details for Peaks
-   Adds view sensor button to results table which opens dialog listing sensors; includes button to site page
-   Makes siteid a link to site page
-   Event populated in map filter form on page load
-   Map filters after navigating from map page
-   Session event displayed on each page
-   Back to Map button displayed on each page
-   Add ability to add, edit and delete Site, Reference Datum and HWM files in Files Tab
-   Add View Results button on map
-   Phone, email and zip code format/validation in site edit form

### Changed

-   Sort result details table by sensor status when opened (Deployed > Retrieved > Proposed > Lost)
-   Revised peak table columns
-   Replaced alerts/confirms with dialog
-   Updated Reference Datum Edit dialog layout
-   Updated radio button/form field layout for all edit dialogs
-   View Details and Edit buttons move within button menu on smaller screens

src/app/app.utilities.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs/Observable';
import { of, throwError } from 'rxjs';
import { Event } from '@interfaces/event';
import { Site } from '@interfaces/site';
import { NetworkName } from '@interfaces/network-name';
import { State } from '@interfaces/state';
import { Sitefullsensors } from '@interfaces/sitefullsensors';
import { SensorType } from '@interfaces/sensor-type';
import { Member } from '@interfaces/member';
import { NoaaStation } from '@interfaces/noaa-station';
import { Role } from '@interfaces/role';
import { Agency } from '@interfaces/agency';

@Injectable()
export class APP_UTILITIES {
    /**
     * Handle Http operation that failed.
     * Let the app continue.
     * @param operation - name of the operation that failed
     * @param result - optional value to return as the observable result
     */
    public static handleError<T>(operation = 'operation', result?: T) {
        return (error: any): Observable<T> => {
            // TODO: send the error to remote logging infrastructure
            console.error(error); // log to console instead

            // TODO: better job of transforming error for user consumption
            // Consider creating a message service for this (https://angular.io/tutorial/toh-pt4)
            console.log(`${operation} failed: ${error.message}`);

            // Let the app keep running by returning an empty result.
            //return of(result as T);
            return throwError(error);
        };
    }
    /**
     * SORT utlity function
     * args: sortArray<array>(array to sort), sortField<string>(field on which to sort),
     * sortDirection<string>('ascend' or 'descend')
     **/
    public static SORT(sortArray, sortField, sortDirection): any {
        if (sortDirection === 'ascend') {
            return sortArray.sort((a, b) =>
                a[sortField] > b[sortField] ? 1 : -1
            );
        }
        if (sortDirection === 'descend') {
            return sortArray.sort((a, b) =>
                a[sortField] < b[sortField] ? 1 : -1
            );
        }
    }

    public static COMPARE(a: number | string, b: number | string, isAsc: boolean) {
        return (a < b ? -1 : 1) * (isAsc ? 1 : -1);
    }
    
    public static FILTER_EVENT(event_name: any, events: Event[]): Event[] {
        if (typeof event_name == 'string') {
            const filterValue = event_name.toLowerCase();
            return events.filter(
                (event) =>
                    event.event_name.toLowerCase().indexOf(filterValue) != -1
            );
        } else {
            const filterValue = event_name.event_name.toLowerCase();
            return events.filter(
                (event) =>
                    event.event_name.toLowerCase().indexOf(filterValue) != -1
            );
        }
    }
    public static FILTER_STATE(state_name: any, states: State[]): State[] {
        //state_name turns into the full state object when a state is selected
        //when you're typing, your text will be matched, when you select a state, the object will be matched
        if (typeof state_name == 'string') {
            const filterValue = state_name.toLowerCase();
            return states.filter(
                (state) =>
                    state.state_name.toLowerCase().indexOf(filterValue) != -1
            );
        } else {
            if (state_name[0] !== undefined) {
                const filterValue = state_name[0].state_name.toLowerCase();
                return states.filter(
                    (state) =>
                        state.state_name.toLowerCase().indexOf(filterValue) != 0
                );
            }
        }
    }
    public static ROUND(num, len): any {
        return Math.round(num * Math.pow(10, len)) / Math.pow(10, len);
    }
    public static SCALE_LOOKUP(mapZoom): any {
        switch (mapZoom) {
            case 19:
                return '1,128';
            case 18:
                return '2,256';
            case 17:
                return '4,513';
            case 16:
                return '9,027';
            case 15:
                return '18,055';
            case 14:
                return '36,111';
            case 13:
                return '72,223';
            case 12:
                return '144,447';
            case 11:
                return '288,895';
            case 10:
                return '577,790';
            case 9:
                return '1,155,581';
            case 8:
                return '2,311,162';
            case 7:
                return '4,622,324';
            case 6:
                return '9,244,649';
            case 5:
                return '18,489,298';
            case 4:
                return '36,978,596';
            case 3:
                return '73,957,193';
            case 2:
                return '147,914,387';
            case 1:
                return '295,828,775';
            case 0:
                return '591,657,550';
        }
    }

    public static FIND_OBJECT_BY_KEY(array, key, value) {
        for (var i = 0; i < array.length; i++) {
            if (array[i][key] === value) {
                return array[i];
            }
        }
        return null;
    }

    public static GET_ROLE() {
        return localStorage.getItem('role');
    }

    // On key down gets value of phone input, sends it to formatter, and then replaces input with formatted value
  public static PHONENUMBERFORMATTER(event: any) {
    if (event !== undefined) {
      const inputid = event.srcElement.id
      const inputField = (<HTMLInputElement>document.getElementById(inputid));
      const formattedInputValue = this.FORMATPHONENUMBER((<HTMLInputElement>document.getElementById(inputid)).value);
      inputField.value = formattedInputValue;
    }
  }

  // formats phone number to display as (111) 111 1111
  public static FORMATPHONENUMBER(value) {
    if (!value) return value;
    const phoneNumber = value.replace(/[^\d]/g, "");
    const phoneNumberLength = phoneNumber.length;
    if (phoneNumberLength < 4) return phoneNumber;
    if (phoneNumberLength < 7) {
      return `(${phoneNumber.slice(0, 3)}) ${phoneNumber.slice(3)}`;
    }
    return `(${phoneNumber.slice(0, 3)}) ${phoneNumber.slice(
      3,
      6
    )}-${phoneNumber.slice(6, 9)}`;
  }

    public static get DUMMY_USER(): Member {
        return {
            member_id: 1,
            fname: 'Bob',
            lname: 'Jones',
            agency_id: 1,
            phone: '(123) 456-7890',
            email: 'stnadmin@usgs.gov',
            role_id: 1,
            username: 'user',
            password: 'password',
            salt: '',
            instrument_status: [],
            events: [],
            peak_summary: [],
            sites: [],
            approvals: [],
            data_file: [],
            hwms: [],
            hwms1: [],
        };
    }

    public static get USERS_DUMMY_DATA_LIST(): Member[] {
        return [
            {
                agency_id: 1,
                approvals: [],
                data_file: [],
                email: 'stnadmin@usgs.gov',
                events: [],
                fname: 'STN',
                hwms: [],
                hwms1: [],
                instrument_status: [],
                lname: 'Admin',
                member_id: 1,
                password: 'password',
                peak_summary: [],
                phone: '(123) 456-7890',
                role_id: 1,
                salt: 'ZTBXfhJn6C3zxD1+',
                sites: [],
                username: 'fradmin',
            },
            {
                agency_id: 1,
                approvals: [],
                data_file: [],
                email: 'billt@usgs.gov',
                emergency_contact_name: '',
                emergency_contact_phone: '',
                events: [],
                fname: 'Bill',
                hwms: [],
                hwms1: [],
                instrument_status: [],
                last_updated: '2018-03-12T21:51:24.157148',
                last_updated_by: 35,
                lname: 'Test',
                member_id: 2,
                password: 'password',
                peak_summary: [],
                phone: '(508) 490-5041',
                role_id: 2,
                salt: '',
                sites: [],
                username: 'billtest'
            }
        ];
    }

    public static get RESULT_DIALOG_DUMMY_DATA() {
        return {
            mapFilterForm: 'event',
            site_id: 54456,
            site_name: 'Site Name',
        };
    }

    public static get EVENTS_DUMMY_DATA_LIST(): Event[] {
        return [
            {
                event_id: 7,
                event_name: 'FEMA 2013 exercise',
                event_start_date: '2013-05-15T04:00:00',
                event_end_date: '2013-05-23T04:00:00',
                event_description: 'Ardent/Sentry 2013 FEMA Exercise',
                event_type_id: 2,
                event_status_id: 2,
                event_coordinator: 36,
                instruments: [],
                hwms: [],
            },
            {
                event_id: 8,
                event_name: 'Wilma',
                event_start_date: '2005-10-20T00:00:00',
                event_end_date: '2005-10-31T00:00:00',
                event_description:
                    'Category 3 in west FL. \nHurricane Wilma was the most intense tropical cyclone ever recorded in the Atlantic basin. Part of the record breaking 2005 Atlantic hurricane season.',
                event_type_id: 2,
                event_status_id: 2,
                event_coordinator: 515,
                instruments: [],
                hwms: [],
            },
        ];
    }

    public static get DUMMY_EVENT(): Event {
        return {
            event_id: 24,
            event_name: 'Sandy',
            event_start_date: '2012-10-21T04:00:00',
            event_end_date: '2012-10-30T04:00:00',
            event_description:
                'historical hurricane data loaded by the data archive team',
            event_type_id: 2,
            event_status_id: 2,
            event_coordinator: 36,
            instruments: [],
            hwms: [],
        };
    }

    public static get DUMMY_SITE_FULL_SENSOR(): Sitefullsensors {
        return {
            sensorType: 'Pressure Transducer',
            deploymentType: 'Water Level',
            instCollection: 'Brackish Water',
            housingType: 'Ruston Steel Pipe',
            sensorBrand: 'Hobo',
            instrument_id: 8567,
            sensor_type_id: 1,
            deployment_type_id: 1,
            location_description:
                'HOBO mounted to southeast head wall of bridge. Same location used for previous deployment',
            serial_number: '9774314',
            interval: 30,
            site_id: 17783,
            event_id: 182,
            inst_collection_id: 2,
            housing_type_id: 0,
            sensor_brand_id: 5,
            vented: 'No',
            instrument_status: [],
            data_files: [],
            files: [],
        };
    }

    public static get SITES_DUMMY_DATA_LIST(): Site[] {
        return [
            {
                site_id: 3149,
                site_no: 'FLIND03149',
                site_name: 'SSS-FL-IND-001WL',
                site_description: 'Sebastian Inlet State Park,',
                address: 'A1A',
                city: 'Sebastian',
                state: 'FL',
                zip: '',
                other_sid: '',
                county: 'Indian River County',
                waterbody: 'Sebastian Inlet intracoastal',
                latitude_dd: 27.85508,
                longitude_dd: -80.45208,
                hdatum_id: 2,
                zone: '',
                is_permanent_housing_installed: '',
                usgs_sid: '',
                noaa_sid: '',
                hcollect_method_id: 1,
                member_id: null,
                site_notes:
                    'Sensor on second piling from land on east side of fishing pier in Sebastian Inlet State Park 0.5 miles west of entrance station south of bridge.',
                safety_notes: '',
                access_granted: '',
                last_updated: '2019-09-10T16:36:00.13484',
                last_updated_by: 3,
                network_name_site: [],
                network_type_site: [],
                objective_points: [],
                instruments: [],
                files: [],
                site_housing: [],
                hwms: [],
            },
            {
                site_id: 3151,
                site_no: 'FLBRE03151',
                site_name: 'SSS-FL-BRE-001WL',
                site_description: 'Lee Wenner Park',
                address: 'King St',
                city: 'Cocoa',
                state: 'FL',
                zip: '32956',
                other_sid: '',
                county: 'Brevard County',
                waterbody: 'Indian River',
                latitude_dd: 28.355,
                longitude_dd: -80.72217,
                hdatum_id: 4,
                zone: '',
                is_permanent_housing_installed: '',
                usgs_sid: '',
                noaa_sid: '',
                hcollect_method_id: 4,
                member_id: null,
                site_notes:
                    'off 520 - 520 E to Riveredge Blvd, turn right and go to Lee Wenner Park, turn left\n\nWooden pilings on T-dock left of boat launch (enter the park and bear left to boat ramp, T-dock is past the ramp',
                safety_notes: '',
                access_granted: '',
                last_updated: '2019-08-29T18:09:32.189622',
                last_updated_by: 105,
                network_name_site: [],
                network_type_site: [],
                objective_points: [],
                instruments: [],
                files: [],
                site_housing: [],
                hwms: [],
            },
        ];
    }

    public static get NETWORK_NAMES_DUMMY_DATA_LIST(): NetworkName[] {
        return [
            {
                network_name_id: 2,
                name: 'SWaTH',
                network_name_site: [],
            },
            {
                network_name_id: 3,
                name: 'SCoRE',
                network_name_site: [],
            },
        ];
    }

    public static get DUMMY_NETWORK_NAME(): NetworkName {
        return {
            network_name_id: 5,
            name: 'Coastal Change Hazards (CCH)',
            network_name_site: [],
        };
    }

    public static get STATES_DUMMY_DATA_LIST(): State[] {
        return [
            {
                state_id: 1,
                state_name: 'Alabama',
                state_abbrev: 'AL',
                counties: [],
            },
            {
                state_id: 7,
                state_name: 'Colorado',
                state_abbrev: 'CO',
                counties: [],
            },
        ];
    }

    public static get DUMMY_STATE(): State {
        return {
            state_id: 13,
            state_name: 'Georgia',
            state_abbrev: 'GA',
            counties: [],
        };
    }

    public static get SENSOR_TYPES_DUMMY_DATA_LIST(): SensorType[] {
        return [
            {
                deploymenttypes: [],
                sensor_type_id: 2,
                sensor: 'Wave Height',
                sensor_deployment: [],
                instruments: [],
            },
            {
                deploymenttypes: [],
                sensor_type_id: 8,
                sensor: 'Water Temperature',
                sensor_deployment: [],
                instruments: [],
            },
        ];
    }

    public static get DUMMY_SENSOR_TYPE(): SensorType {
        return {
            deploymenttypes: [],
            sensor_type_id: 4,
            sensor: 'Webcam',
            sensor_deployment: [],
            instruments: [],
        };
    }

    public static get FILTERED_SITES_SAMPLE_QUERY_PARAMS(): string {
        return `Event=&State=MS&SensorType=&NetworkName=&OPDefined=&HWMOnly=&HWMSurveyed=&SensorOnly=&RDGOnly=&HousingTypeOne=1`;
    }

    public static get NOAA_DUMMY_DATA_LIST(): NoaaStation {
        return {
            forecast: false,
            greatlakes: false,
            id: '8665530',
            lat: 32.780833,
            lng: -79.923611,
            name: "Charleston, Cooper River Entrance",
            observedst: true,
            portscode: 'ch',
            shefcode: "CHTS1",
            state: 'SC',
            stormsurge: false,
            tidal: true,
            tideType: 'Mixed',
            timezone: 'EST',
            timezonecorr: -5,
        };
    }

    public static get ROLES_DUMMY_DATA_LIST(): Role[] {
        return [
            {
                role_id: 1,
                role_name: "Admin",
                role_description: "Administrator",
                members: []
            },
            {
                role_id: 2,
                role_name: "Manager",
                role_description: "Manager (web page, provisional data)",
                members: []
            },
            {
                role_id: 3,
                role_name: "Field",
                role_description: "Field data entry",
                members: []
            },
            {
                role_id: 4,
                role_name: "Public",
                role_description: "Read-only (web page only)",
                members: []
            }
        ];
    }
    public static get AGENCY_DUMMY_DATA_LIST(): Agency[] {
        return [
            {
                agency_id: 1,
                agency_name: "USGS MA-RI",
                address: "10 Bearfoot Rd",
                city: "Northborough",
                state: "MA",
                zip: "1532",
                phone: "(000) 000-0000",
                Links: [
                    {
                        rel: "self",
                        Href: "https://stn.wim.usgs.gov/STNServices/Agencies/1",
                        method: "GET"
                    },
                    {
                        rel: "edit",
                        Href: "https://stn.wim.usgs.gov/STNServices/Agencies/1",
                        method: "PUT"
                    },
                    {
                        rel: "delete",
                        Href: "https://stn.wim.usgs.gov/STNServices/Agencies/1",
                        method: "DELETE"
                    }
                ]
            },
            {
                agency_id: 2,
                agency_name: "USGS FL",
                address: "7500 SW 36th Street",
                city: "Davie",
                state: "FL",
                zip: "33314",
                phone: "(000) 000-0000",
                Links: [
                    {
                        rel: "self",
                        Href: "https://stn.wim.usgs.gov/STNServices/Agencies/2",
                        method: "GET"
                    },
                    {
                        rel: "edit",
                        Href: "https://stn.wim.usgs.gov/STNServices/Agencies/2",
                        method: "PUT"
                    },
                    {
                        rel: "delete",
                        Href: "https://stn.wim.usgs.gov/STNServices/Agencies/2",
                        method: "DELETE"
                    }
                ]
            }
        ];
    }
}
